src/command_func.h

```diff
@@ -37,6 +37,13 @@ static const CommandCost CMD_ERROR = CommandCost(INVALID_STRING_ID);
  */
 #define return_cmd_error(errcode) return CommandCost(errcode);
 
+/*
+ * Returns from a function if the provided command result is a failure.
+ *
+ * @param command_cost The result of the command to check
+ */
+#define return_on_cmd_failure(command_result) if (command_result.Failed()) return command_result;
```

Comment:
This feels at best suboptimal, at worst possibly using a temporary incorrectly. Currently it would expand to (e.g.)

```
if (CheckTileOwnership(tile).Failed()) return CheckTileOwnership(tile);

It's only used a few times, I'd just expand them all out (reusing the existing `cost` variable should be fine?)

Reply:
And just in favour of not adding macros generally, tbh.

Reply:
I just wanted to create a convenience macro to make this a little easier without having to deal with naming a local variable, potentially running the risk over overwriting something. But I agree that my implementation was a little naive. I came up with this alternative:

`#define return_on_cmd_failure(command) { auto return_on_cmd_failure_result = command; if (return_on_cmd_failure_result.Failed()) return return_on_cmd_failure_result; }`

But maybe I'm just overcomplicating things. You guys make the call, I'm fine with both fixing the macro or removing it altogether.

Reply:
Slept on it a little and I decided to remove it. Too little gain. I've just written them out like you see elsewhere in the code.